### PR TITLE
docs(dp): add section on how to resize kuma-sidecar containers on Kubernetes

### PR DESCRIPTION
**Pull request by Automaat**
Fixes: #994 

Signed-off-by: Marcin Skalski <marcin.skalski@konghq.com>

### CODE REVIEW

File: app/docs/dev/documentation/requirements.md [hunk 1]
```diff
@@ -34,3 +34,82 @@ To see if you may need to increase your control-plane's spec, there are two main
 For any large mesh using transparent-proxy it's highly recommended to use [reachable-services](/docs/{{ page.version }}/networking/transparent-proxying#reachable-services).
 
 You can also find tuning configuration in the [fine-tuning](/docs/{{ page.version }}/documentation/fine-tuning) section of the docs.
+
+## Sizing your dataplane container on Kubernetes
```
Comment (michaelbeaumont) [inline app/docs/dev/documentation/requirements.md:38]:
```suggestion
## Sizing your sidecar container on Kubernetes
```

File: app/docs/dev/documentation/requirements.md [hunk 2]
```diff
@@ -34,3 +34,82 @@ To see if you may need to increase your control-plane's spec, there are two main
...
+
+When deploying Kuma on Kubernetes, dataplane is deployed as a separate container in Pod. By default, `kuma-sidecar` container is configured like this:
```
Comment (michaelbeaumont) [inline app/docs/dev/documentation/requirements.md:40]:
```suggestion
When deploying Kuma on Kubernetes, the sidecar is deployed as a separate container, `kuma-sidecar`, in your `Pods`. By default it has the following resource requests and limits:
```

File: app/docs/dev/documentation/requirements.md [hunk 3]
```diff
@@ -34,3 +34,82 @@ To see if you may need to increase your control-plane's spec, there are two main
...
+resources:
+    requests:
+        cpu: 50m
+        memory: 64Mi
+    limits:
+        cpu: 500m
+        memory: 512Mi
+```
+
+This resources configuration should be enough for most use cases. In some cases like when you cannot scale horizontally, or your service handle lots of concurrent traffic, you may need to change those values. You can do it using [ContainerPatch resource](/docs/{{ page.version }}/explore/dpp-on-kubernetes/#custom-container-configuration).
```
Comment (michaelbeaumont) [inline app/docs/dev/documentation/requirements.md:52]:
```suggestion
This configuration should be enough for most use cases. In some cases, like when you cannot scale horizontally or your service handles lots of concurrent traffic, you may need to change these values. You can do this using the [`ContainerPatch` resource](/docs/{{ page.version }}/explore/dpp-on-kubernetes/#custom-container-configuration). 
```

File: app/docs/dev/documentation/requirements.md [hunk 4]
```diff
@@ -34,3 +34,82 @@ To see if you may need to increase your control-plane's spec, there are two main
...
+        cpu: 50m
+        memory: 64Mi
+    limits:
+        cpu: 500m
+        memory: 512Mi
+```
+
+This resources configuration should be enough for most use cases. In some cases like when you cannot scale horizontally, or your service handle lots of concurrent traffic, you may need to change those values. You can do it using [ContainerPatch resource](/docs/{{ page.version }}/explore/dpp-on-kubernetes/#custom-container-configuration). 
+
+To change thoes resources, you need `ContainerPatch` with `sidecarPatch`. You have multiple options. You can modify single parameter of resource section:
```
Comment (michaelbeaumont) [inline app/docs/dev/documentation/requirements.md:54]:
```suggestion

For example, you can modify individual parameters under `resources`:
```

File: app/docs/dev/documentation/requirements.md [hunk 5]
```diff
@@ -34,3 +34,82 @@ To see if you may need to increase your control-plane's spec, there are two main
...
+  name: container-patch-1
+  namespace: kuma-system
+spec:
+  sidecarPatch:
+    - op: add
+      path: /resources/requests/cpu
+      value: '"1"'
+```
+
+You can modify whole `limits` section:
```
Comment (michaelbeaumont) [inline app/docs/dev/documentation/requirements.md:69]:
```suggestion
you could modify the entire `limits`, `request` or `resources` sections:
```

File: app/docs/dev/documentation/requirements.md [hunk 6]
```diff
@@ -34,3 +34,82 @@ To see if you may need to increase your control-plane's spec, there are two main
...
+        "limits": {
+            "cpu": "1",
+            "memory": "1G"
+        },
+        "requests": {
+            "cpu": "500m",
+            "memory": "500Mi"
+        }
+    }'
+```
```
Comment (michaelbeaumont) [inline app/docs/dev/documentation/requirements.md:109]: IMO this is too many different examples. As soon as you understand JSON patch it becomes unnecessary to show this much

File: app/docs/dev/documentation/requirements.md [hunk 7]
```diff
@@ -34,3 +34,82 @@ To see if you may need to increase your control-plane's spec, there are two main
...
+
+Precise documentation on how to apply those `ContainerPatch` resources to specific Pod can be found [here](/docs/{{ page.version }}/explore/dpp-on-kubernetes/#workload-matching).
```
Comment (michaelbeaumont) [inline app/docs/dev/documentation/requirements.md:111]:
```suggestion
Check [the `ContainerPatch` documentation](/docs/{{ page.version }}/explore/dpp-on-kubernetes/#workload-matching) for how to apply these resources to specific `Pods`.
```

File: app/docs/dev/documentation/requirements.md [hunk 8]
```diff
@@ -34,3 +34,82 @@ To see if you may need to increase your control-plane's spec, there are two main
...
+
+{% tip %}
+**Note**: When changing those resources, remember that they must be described using [Kubernetes resource units](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes)
```
Comment (michaelbeaumont) [inline app/docs/dev/documentation/requirements.md:114]:
```suggestion
**Note**: When changing these resources, remember that they must be described using [Kubernetes resource units](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes)
```

File: app/docs/dev/documentation/requirements.md (hunk 6 above)
Comment (Automaat) [reply app/docs/dev/documentation/requirements.md]: I was thinking about this when I added those, and probably you are right, removed the third example

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**